src/libvirtApi/network.js

```diff
@@ -62,64 +62,61 @@ export function networkDeactivate({ connectionName, objPath }) {
+        throw ex;
+    }
 }
 
 /*
  * Read properties of a single Network
  *
  * @param Network object path
  */
-export function networkGet({
+export async function networkGet({
     id: objPath,
     connectionName,
     updateOnly,
 }) {
-    const props = {};
-
-    return call(connectionName, objPath, 'org.freedesktop.DBus.Properties', 'GetAll', ['org.libvirt.Network'], { timeout, type: 's' })
-            .then(resultProps => {
-                /* Sometimes not all properties are returned; for example when some network got deleted while part
-                 * of the properties got fetched from libvirt. Make sure that there is check before reading the attributes.
-                 */
-                if ("Active" in resultProps[0])
-                    props.active = resultProps[0].Active.v.v;
-                if ("Persistent" in resultProps[0])
-                    props.persistent = resultProps[0].Persistent.v.v;
-                if ("Autostart" in resultProps[0])
-                    props.autostart = resultProps[0].Autostart.v.v;
-                if ("Name" in resultProps[0])
-                    props.name = resultProps[0].Name.v.v;
-                props.id = objPath;
-                props.connectionName = connectionName;
-
-                return call(connectionName, objPath, 'org.libvirt.Network', 'GetXMLDesc', [0], { timeout, type: 'u' });
-            })
-            .then(xml => {
-                const network = parseNetDumpxml(xml);
-                return store.dispatch(updateOrAddNetwork(Object.assign({}, props, network), updateOnly));
-            })
-            .catch(ex => console.warn('GET_NETWORK action failed for path', objPath, ex.toString()));
+    try {
+        const props = {};
+
+        const [resultProps] = await call(connectionName, objPath, 'org.freedesktop.DBus.Properties', 'GetAll', ['org.libvirt.Network'], { timeout, type: 's' });
+        /* Sometimes not all properties are returned; for example when some network got deleted while part
+            * of the properties got fetched from libvirt. Make sure that there is check before reading the attributes.
+            */
+        if ("Active" in resultProps)
+            props.active = resultProps.Active.v.v;
+        if ("Persistent" in resultProps)
+            props.persistent = resultProps.Persistent.v.v;
+        if ("Autostart" in resultProps)
+            props.autostart = resultProps.Autostart.v.v;
+        if ("Name" in resultProps)
+            props.name = resultProps.Name.v.v;
+        props.id = objPath;
+        props.connectionName = connectionName;
+
+        const [xml] = await call(connectionName, objPath, 'org.libvirt.Network', 'GetXMLDesc', [0], { timeout, type: 'u' });
+        const network = parseNetDumpxml(xml);
+        store.dispatch(updateOrAddNetwork(Object.assign(props, network), updateOnly));
+    } catch (ex) {
+        console.warn('GET_NETWORK action failed for path', objPath, ex.toString());
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1400-20240125-053137-bfbdf98b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)